server/pbenchinacan/deploy

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash -e
+
+#
+# This script deploys a Pbench Server instance from a container image.
+#
+# Steps performed:
+#   - The Pbench Dashboard application files are copied from the build directory
+#     into the SRV_PBENCH file system.  (If the build directory does not exist,
+#     an empty directory is created at the input and output locations, a warning
+#     is printed, and the Pbench Server is deployed without a Dashboard.)
+#   - If present in the current directory, the Nginx favicon file is copied into
+#     the appropriate location in the SRV_PBENCH file system.
+#   - The ownership of the files in the SRV_PBENCH file system will are set for
+#     access inside the container.
+#   - The Pbench Server container is started and run as a detached process.
+#
+# The key inputs are the following environment variables:
+#   - PB_DEPLOY_FILES:  the location (on the host) of the configuration files
+#                       for the Pbench Server and rsyslog.
+#   - SRV_PBENCH:  the location of the file system (on the host) which will be
+#                       mounted inside the container at /srv/pbench.
+#
+# Other inputs which can be provided by environment variables include:
+#   - the name and pull policy for the Pbench Server container image
+#   - the location of the Pbench Dashboard app files and the Nginx Favicon
+#   - the location of the Pbench Git checkout (used to locate the above)
+#
+# In all cases, suitable defaults are provided for any value which is undefined.
+#
+
+#+
+# Configuration
+#-
+
+# Pbench Server container image
+${PB_SERVER_IMAGE:="images.paas.redhat.com/pbench/pbench-server:main"}
+${PB_SERVER_IMAGE_PULL_POLICY:="newer"}
+
+# Locations on the host
+${PB_DASHBOARD_DIR:=${GITTOP:=$(git rev-parse --show-toplevel)}/dashboard/build}
+${NGINX_FAVICON:=${GITTOP}/server/lib/config/color-square.ico}
+${PB_DEPLOY_FILES:=${HOME}/Deploy}
+${SRV_PBENCH:=/srv/pbench}
+
+#+
+# Deployment
+#-
+
+# Update the dashboard code, removing any existing code and copying in fresh.
+# We expect that the host dashboard directory already exists and contains a
+# build of the Pbench Dashboard application; however, if it is missing, we
+# deploy the Pbench Server without a Dashboard deployment.
+podman unshare rm -r ${SRV_PBENCH}/public_html/dashboard
+if [[ -d "${PB_DASHBOARD_DIR}" ]]; then
+    podman unshare cp -r ${PB_DASHBOARD_DIR} ${SRV_PBENCH}/public_html/dashboard
+else
+    echo "Warning:  deploying the Pbench Server without a Dashboard" >&2
+    podman unshare mkdir -m 0755 -p ${SRV_PBENCH}/public_html/dashboard
+fi
```

Comment:
Here's a thought, though perhaps `run-pbench-in-a-can` is a better home ... should we be trying to *build* a dashboard from source as part of routine deployment? We'll need to do this in the CI anyway at some point, and it might be handy for routine testing...

Reply:
I concur that some of this logic might be better placed in `run-pbench-in-a-can`, but I will assert (and I _think_ @portante would concur) that _building_ the Dashboard should not be part of _deployment_.  Nevertheless, it probably would be good to build the Dashboard as one of the upstream prep tasks, and I'll look for an appropriate slot for that.

Reply:
I'm going to defer the auto-build/deployment of the Dashboard to a subsequent PR.